src/main/java/org/team1540/robot2025/subsystems/drive/ModuleIOTalonFX.java

```diff
@@ -0,0 +1,255 @@
+        drive = new TalonFX(constants.DriveMotorId, DrivetrainConstants.CAN_BUS);
+        turn = new TalonFX(constants.SteerMotorId, DrivetrainConstants.CAN_BUS);
+        cancoder = new CANcoder(constants.EncoderId, DrivetrainConstants.CAN_BUS);
+
+        driveConfig = hw.driveConfig();
+        turnConfig = hw.turnConfig();
+        cancoderConfig = hw.turnEncoderConfig();
+
+        lastDriveConfigStatus =
+                PhoenixUtil.tryUntilOk(5, () -> drive.getConfigurator().apply(driveConfig));
+        lastTurnConfigStatus =
+                PhoenixUtil.tryUntilOk(5, () -> turn.getConfigurator().apply(turnConfig));
+        lastCancoderConfigStatus =
+                PhoenixUtil.tryUntilOk(5, () -> cancoder.getConfigurator().apply(cancoderConfig));
+
+        // Create timestamp queue
+        timestampQueue = OdometryThread.getInstance().makeTimestampQueue();
+
+        // Create drive status signals
+        drivePosition = drive.getPosition();
+        drivePositionQueue = OdometryThread.getInstance().registerSignal(drive.getPosition());
+        driveVelocity = drive.getVelocity();
+        driveAppliedVolts = drive.getMotorVoltage();
+        driveSupplyCurrent = drive.getSupplyCurrent();
+        driveStatorCurrent = drive.getStatorCurrent();
+        driveTemp = drive.getDeviceTemp();
+
+        // Create turn status signals
+        turnAbsolutePosition = cancoder.getAbsolutePosition();
+        turnPosition = turn.getPosition();
+        turnPositionQueue = OdometryThread.getInstance().registerSignal(turn.getPosition());
+        turnVelocity = turn.getVelocity();
+        turnAppliedVolts = turn.getMotorVoltage();
+        turnSupplyCurrent = turn.getSupplyCurrent();
+        turnStatorCurrent = turn.getStatorCurrent();
+        turnTemp = turn.getDeviceTemp();
+
+        // Configure periodic frames
+        BaseStatusSignal.setUpdateFrequencyForAll(DrivetrainConstants.ODOMETRY_FREQUENCY, drivePosition, turnPosition);
+        BaseStatusSignal.setUpdateFrequencyForAll(
+                50.0,
+                driveVelocity,
+                driveAppliedVolts,
+                driveSupplyCurrent,
+                driveStatorCurrent,
+                driveTemp,
+                turnAbsolutePosition,
+                turnVelocity,
+                turnAppliedVolts,
+                turnSupplyCurrent,
+                turnStatorCurrent,
+                turnTemp);
+        ParentDevice.optimizeBusUtilizationForAll(drive, turn, cancoder);
+    }
+
+    @Override
+    public void updateInputs(ModuleIOInputs inputs) {
+        // Refresh all signals
+        StatusCode driveStatus = BaseStatusSignal.refreshAll(
+                drivePosition, driveVelocity, driveAppliedVolts, driveStatorCurrent, driveTemp);
+        StatusCode turnStatus =
+                BaseStatusSignal.refreshAll(turnPosition, turnVelocity, turnAppliedVolts, turnStatorCurrent, driveTemp);
```

Comment:
isn't doing it all at once faster or less blocking or something

Reply:
Not in a way that's noticeable here. It's also nice to be able to get the connection status of the motor in one go like this.